lib/sync/sync.ts

```diff
@@ -16,55 +16,80 @@
 
 import {
     GitProject,
+    HandlerResult,
     logger,
     ProjectFile,
 } from "@atomist/automation-client";
 import {
-    fakeContext,
+    CommandHandlerRegistration,
+    CommandListenerInvocation,
     ProjectLoadingParameters,
-    SoftwareDeliveryMachine,
 } from "@atomist/sdm";
 import * as _ from "lodash";
 import { KubernetesSyncOptions } from "../config";
 import { parseKubernetesSpecFile } from "../deploy/spec";
 import { applySpec } from "../kubernetes/apply";
 import { decryptSecret } from "../kubernetes/secret";
 import { errMsg } from "../support/error";
+import { clientName } from "../support/name";
 import { defaultCloneOptions } from "./clone";
 import { k8sSpecGlob } from "./diff";
+import { isRemoteRepo } from "./repo";
+
+/**
+ * Command to synchronize the resources in a Kubernetes cluster with
+ * the resource specs in the configured sync repo.  The sync repo will
+ * be cloned and the resources applied against the Kubernetes API in
+ * lexical order sorted by file name.  If no sync repo is configured
+ * in the SDM, the command errors.  This command is typically executed
+ * on an interval timer by setting the `intervalMinutes`
+ * [[KubernetesSyncOptions]].
+ */
+export const kubernetesSync: CommandHandlerRegistration = {
+    intent: `kube sync ${clientName()}`,
+    name: "KubernetesSync",
+    listener: repoSync,
+};
 
 /**
  * Clone the sync repo and apply the specs to the Kubernetes cluster.
  */
-export async function repoSync(sdm: SoftwareDeliveryMachine): Promise<void> {
-    const disposers: Array<() => Promise<void>> = [];
-    const workspaceId: string = _.get(sdm, "configuration.workspaceIds[0]");
-    const context = fakeContext(workspaceId);
-    context.lifecycle = {
-        dispose: async () => { await Promise.all(disposers.map(d => d())); },
-        registerDisposable: (d: () => Promise<void>) => { disposers.push(d); },
-    };
+export async function repoSync(cli: CommandListenerInvocation): Promise<HandlerResult> {
+    const opts: KubernetesSyncOptions = _.get(cli.configuration, "sdm.k8s.options.sync");
+    if (!opts) {
```

Comment:
I’d move that check before the createJob call to prevent job creation for cases where there are no sync options defined.

Reply:
It's there too, but I'll keep it here too since there are two ways into this method.